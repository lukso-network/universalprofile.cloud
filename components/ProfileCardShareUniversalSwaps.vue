<script setup lang="ts">
type Props = {
  profile?: Profile
}

const props = defineProps<Props>()

const handleOpen = () => {
  window.open(universalSwapsProfileUrl(props.profile?.address), '_blank')
}
</script>

<template>
  <lukso-tooltip variant="white" text="universalswaps.io" placement="left">
    <lukso-button
      size="small"
      variant="secondary"
      custom-class="px-0 overflow-hidden size-full"
      class="group size-[30px]"
      @click="handleOpen"
    >
      <img
        src="/images/universalswaps-logo.png"
        alt=""
        class="max-h-4 max-w-[20px] grayscale transition group-hover:grayscale-0"
      />
    </lukso-button>
  </lukso-tooltip>
</template>
