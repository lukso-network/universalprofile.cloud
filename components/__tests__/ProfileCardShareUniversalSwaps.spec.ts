import { renderSuspended } from '@nuxt/test-utils/runtime'
import { screen } from '@testing-library/vue'
import { describe, expect, it, vi } from 'vitest'

import ProfileCardShareUniversalSwaps from '@/components/ProfileCardShareUniversalSwaps.vue'

describe('ProfileCardShareUniversalSwaps', () => {
  const profile: Profile = {
    address: '0x1234567890abcdef1234567890abcdef12345678',
  }

  it('should render', async () => {
    const component = await renderSuspended(ProfileCardShareUniversalSwaps, {
      props: { profile },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should open in a new tab', async () => {
    const component = await renderSuspended(ProfileCardShareUniversalSwaps, {
      props: { profile },
    })

    const windowOpenSpy = vi.spyOn(window, 'open')

    ;(await component.findByRole('img')).click()

    expect(windowOpenSpy).toHaveBeenCalledWith(
      'https://universalswaps.io/social/0x1234567890abcdef1234567890abcdef12345678',
      '_blank'
    )
  })
})
