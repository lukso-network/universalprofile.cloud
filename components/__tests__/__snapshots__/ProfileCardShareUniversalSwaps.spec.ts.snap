// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileCardShareUniversalSwaps > should render 1`] = `
"<div id="test-wrapper">
  <lukso-tooltip variant="white" text="universalswaps.io" placement="left">
    <lukso-button size="small" variant="secondary" custom-class="px-0 overflow-hidden size-full" class="group size-[30px]"><img src="/images/universalswaps-logo.png" alt="" class="max-h-4 max-w-[20px] grayscale transition group-hover:grayscale-0"></lukso-button>
  </lukso-tooltip>
</div>"
`;
